core/chaincode/lifecycle/lifecycle.go

```diff
@@ -424,7 +424,23 @@ func (ef *ExternalFunctions) ApproveChaincodeDefinitionForOrg(chname, ccname str
 			}
 
 			if err := uncommittedParameters.Equal(cd.Parameters()); err == nil {
-				return errors.Errorf("attempted to redefine uncommitted sequence (%d) for namespace %s with unchanged content", requestedSequence, ccname)
+				// also check package ID updates
+				uncommittedPackageID := ""
+				metadata, ok, err := ef.Resources.Serializer.DeserializeMetadata(ChaincodeSourcesName, privateName, orgState)
+				if err != nil {
+					return errors.WithMessagef(err, "could not deserialize chaincode-source metadata for %s", privateName)
+				}
+				if ok {
+					ccLocalPackage := &ChaincodeLocalPackage{}
+					if err := ef.Resources.Serializer.Deserialize(ChaincodeSourcesName, privateName, metadata, ccLocalPackage, orgState); err != nil {
+						return errors.WithMessagef(err, "could not deserialize chaincode package for %s", privateName)
+					}
+					uncommittedPackageID = ccLocalPackage.PackageID
+				}
+
+				if uncommittedPackageID == packageID {
+					return errors.Errorf("attempted to redefine uncommitted sequence (%d) for namespace %s with unchanged content", requestedSequence, ccname)
+				}
```

Comment:
This works as-is but it feels like we could simplify this down and avoid the uncommittedPackageID variable if we just move the check into the "ok" block where we're setting the uncommittedPackageID: 

```
if ok {
        ccLocalPackage := &ChaincodeLocalPackage{}
	if err := ef.Resources.Serializer.Deserialize(ChaincodeSourcesName, privateName, metadata, ccLocalPackage, orgState); err != nil {
	         return errors.WithMessagef(err, "could not deserialize chaincode package for %s", privateName)
	}
	if ccLocalPackage.PackageID == packageID {
                 return errors.Errorf("attempted to redefine uncommitted sequence (%d) for namespace %s with unchanged content", requestedSequence, ccname)
        }
}
```